File: Python/procedural/control_structure/mix/guessTheNumber.py
#Brief: Guess the number
#Guess the number
#- The program has a random number between 1 and 100
#- The user has to guess the number
#- The program tells the user if the number is higher or lower
#- Number of attempts: 10 times
# Date: 20/08/2020
# Version 1.0


import random as rd

LOWER_LIMIT = 1
HIGH_LIMIT = 100
RANDON_NUMBER = rd.randint(LOWER_LIMIT, HIGH_LIMIT)  
ATTEMPTS = 10

user_number = int(input("Guess a number between 1 and 100: "))

if user_number < LOWER_LIMIT or user_number > HIGH_LIMIT:
    print("The number is out of range")
else:
    for i in range(ATTEMPTS):
        if user_number > RANDON_NUMBER:
            print("The number is less")
        elif user_number < RANDON_NUMBER:
            print("The number is greater")
        else:
            print("Congratulations, you guessed the number")
            break
        user_number = int(input("Guess a number between 1 and 100: "))
    print("You have run out of attempts")

File: C#/procedure/guessTheNumber.cs
/*Brief: Write a program that generates a random number between 1 and 100. 
        The user must try to guess it.
 Date: 03/12/2024
 Version: 1.0
*/

using System;

// Generate a random number between 1 and 100
Random random = new Random();
int secretNumber = random.Next(1, 101);

Console.WriteLine("Welcome to the Number Guessing Game!");
Console.WriteLine("You have 5 attempts to guess a number between 1 and 100.");

int attempts = 0;
bool guessed = false;

// Loop for up to 5 attempts
while (attempts < 5 && !guessed)
{
    Console.Write($"Attempt {attempts + 1}: Enter your guess: ");
    string input = Console.ReadLine();

    // Try to parse the input into a number
    if (int.TryParse(input, out int userGuess))
    {
        attempts++;
        if (userGuess == secretNumber)
        {
            Console.WriteLine($"Congratulations! You guessed the secret number ({secretNumber}) in {attempts} attempts.");
            guessed = true;
        }
        else if (userGuess < secretNumber)
        {
            Console.WriteLine("The number is higher.");
        }
        else
        {
            Console.WriteLine("The number is lower.");
        }
    }
    else
    {
        Console.WriteLine("Please enter a valid number.");
    }
}

if (!guessed)
{
    Console.WriteLine($"Sorry, you've run out of attempts. The secret number was: {secretNumber}");
}

Console.WriteLine("Thanks for playing!");
